httpclient-parent/httpclient/src/test/java/io/yupiik/uship/httpclient/core/listener/impl/HARDumperListenerTest.java

```diff
@@ -0,0 +1,147 @@
+import java.net.http.HttpClient;
+import java.net.http.HttpHeaders;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.util.Map;
+import java.util.Optional;
+import java.util.logging.Logger;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class HARDumperListenerTest {
+
+    @Test
+    void harJson(@TempDir final Path temp) throws Exception {
+        final var output = temp.resolve("harJson.json");
+        final String expected = "{\n" + "  \"log\":{\n" + "    \"comment\":\"\",\n"
+            + "    \"entries\":[\n" + "      {\n" + "        \"request\":{\n"
+            + "          \"bodySize\":-1,\n" + "          \"comment\":\"\",\n"
+            + "          \"headerSize\":0,\n" + "          \"headers\":[\n" + "          ],\n"
+            + "          \"httpVersion\":\"HTTP/1.1\",\n" + "          \"method\":\"GET\",\n"
+            + "          \"url\":\"http://localhost:1234/test1\"\n" + "        },\n"
+            + "        \"response\":{\n" + "          \"bodySize\":0,\n"
+            + "          \"comment\":\"\",\n" + "          \"content\":{\n"
+            + "            \"compression\":0,\n" + "            \"size\":0\n" + "          },\n"
+            + "          \"headers\":[\n" + "          ],\n" + "          \"headersSize\":0,\n"
+            + "          \"httpVersion\":\"HTTP/1.1\",\n" + "          \"status\":200,\n"
+            + "          \"statusText\":\"OK\"\n" + "        },\n"
+            + "        \"startedDateTime\":\"1970-01-01T00:00:00.000Z\",\n" + "        \"time\":1\n"
+            + "      }\n" + "    ],\n" + "    \"version\":\"1.2\"\n" + "  }\n" + "}";
+
+        var clock = new Clock() {
+            private Instant instant = Instant.EPOCH;
+            @Override
+            public ZoneId getZone() {
+                return ZoneId.of("UTC");
+            }
+
+            @Override
+            public Clock withZone(ZoneId zone) {
+                return this;
+            }
+
+            @Override
+            public Instant instant() {
+                return instant;
+            }
+
+            public void plusMillis(long value) {
+                instant = instant.plusMillis(value);
+            }
+        };
+        final HARDumperListener.Configuration configuration = new HARDumperListener.Configuration(
+            output,
+            clock,
+            Logger.getLogger("test"));
+
+        try (final var listener = new HARDumperListener(configuration)) {
+
+            final var request = requestGet();
+            final var before = listener.before(1, request);
+            clock.plusMillis(1);
+            listener.after(before.state(), before.request(), null, response(200, request));
+        }
+
+        assertEquals(expected, Files.readString(output));
+    }
+
+    //TODO move to test module?
```

Comment:
we can make it a constant but it is not bad to keep it encapsulated in tests for now (= if you want you can drop the todo)

Reply:
ok